File: 兴趣班入门百练讲义/NQ100/NQ038/NQ038_improved.py
# NQ038 统计数字字符个数
def main():
    import sys
    input_lines = [line.rstrip('\n') for line in sys.stdin]
    ptr = 0
    test_case_num = int(input_lines[ptr])
    ptr += 1
    
    for _ in range(test_case_num):
        if ptr >= len(input_lines):
            break
        input_line = input_lines[ptr]
        ptr += 1
        
        digit_count = sum(1 for c in input_line if c.isdigit())  # 统计数字字符个数
        print(digit_count)

if __name__ == "__main__":
    main()

File: 兴趣班入门百练讲义/NQ100/NQ038/NQ038_improved.cpp
// NQ038 统计数字字符个数
#include <cctype>
#include <iostream>
#include <string>
using namespace std;

int main() {
    int test_case_num;
    cin >> test_case_num;
    cin.ignore();  // 清除换行符
    
    while (test_case_num--) {
        string input_line;
        getline(cin, input_line);
        
        int digit_count = 0;
        for (char c : input_line) {
            if (isdigit(static_cast<unsigned char>(c))) digit_count++;  // 统计数字字符
        }
        
        cout << digit_count << endl;
    }
    
    return 0;
}